### ISSUE TITLE
Push zip to an endpoint

### ISSUE THREAD

**Issue by RealHarshThakur** [sharkyze/waypoint-plugin-archive#3]
Great plugin ! This can be used in many places but I think most common use case would be to upload the archives to an endpoint. Do you have any plan on adding that?

**Comment by xsaamiir** [comment]
Hey @harshthakur9030, thank you so much for your feedback.

Do you have an example of where you would want to upload the archive ?

I have a [separate registry plugin to upload the archive to Google Cloud Storage](https://github.com/sharkyze/waypoint-plugin-archive). I preferred extracting the [`registry`](https://www.waypointproject.io/docs/lifecycle/build#registry) step to a separate plugin since I imagined most users would have different uses for the output archive.

**Comment by RealHarshThakur** [comment]
That's true! I have seen other repos using something like this: https://github.com/graymeta/stow . Not sure if this is the best but definitely promising and serves the purpose. Has this been proposed to upstream Waypoint? 
PS: Would definitely like to contribute if i can

**Comment by xsaamiir** [comment]
That's a great reference, thanks for sharing, it reminds of this project: https://gocloud.dev/howto/blob.
That could be a good abstraction to build an uploader compatible with several cloud storage services.

I haven't proposed this to upstream Waypoint, I am not sure if it would be really useful to them although I saw they might work on something similar soon: https://github.com/hashicorp/waypoint/issues/650

**Comment by RealHarshThakur** [comment]
I think you should open up a thread and propose it. :)
